src/utils/components/EnvironmentEditor/components/EnvironmentEditor.tsx

```diff
@@ -89,63 +90,78 @@ const EnvironmentEditor: React.FC<EnvironmentEditorProps> = ({
   return (
     <div className="row pairs-list__row">
       <div className="col-xs-5 pairs-list__value-pair-field">
-        <Select
-          onSelect={(event, selection: EnvironmentOption) => {
-            onChange(diskName, selection.getName(), serial, selection.getKind());
-            setOpen(false);
-          }}
-          toggleIcon={
-            kind ? (
-              <span className={`co-m-resource-icon co-m-resource-${kind}`}>
-                {MapKindToAbbr[kind]}
-              </span>
-            ) : null
-          }
-          aria-labelledby="environment-name-header"
-          hasInlineFilter
-          isOpen={isOpen}
-          maxHeight={400}
-          menuAppendTo="parent"
-          onFilter={onFilter}
-          onToggle={(_, isExpanded) => setOpen(isExpanded)}
-          placeholderText={t('Select a resource')}
-          selections={new EnvironmentOption(environmentName, kind)}
-          variant={SelectVariant.single}
-        >
-          <SelectGroup key="group1" label={t('Secrets')}>
-            {secrets.map((secret) => (
-              <EnvironmentSelectOption
-                isDisabled={environmentNamesSelected?.includes(secret.metadata.name)}
-                key={secret.metadata.name}
-                kind={EnvironmentKind.secret}
-                name={secret.metadata.name}
-              />
-            ))}
-          </SelectGroup>
-          <Divider key="divider1" />
-          <SelectGroup key="group2" label={t('Config Maps')}>
-            {configMaps.map((configMap) => (
-              <EnvironmentSelectOption
-                isDisabled={environmentNamesSelected?.includes(configMap.metadata.name)}
-                key={configMap.metadata.name}
-                kind={EnvironmentKind.configMap}
-                name={configMap.metadata.name}
-              />
-            ))}
-          </SelectGroup>
-          <Divider key="divider2" />
-          <SelectGroup key="group3" label={t('Service Accounts')}>
-            {serviceAccounts.map((serviceAccount) => (
-              <EnvironmentSelectOption
-                isDisabled={environmentNamesSelected?.includes(serviceAccount.metadata.name)}
-                key={serviceAccount.metadata.name}
-                kind={EnvironmentKind.serviceAccount}
-                name={serviceAccount.metadata.name}
-              />
-            ))}
-          </SelectGroup>
-        </Select>
+        {loaded ? (
+          <Select
```

Comment:
Can you please extract this Select to a different component or if it's too tricky with prop drilling any other way to keep the file shorter?

Reply:
yes sure